WordPress/Classes/Services/PostService.m

```diff
@@ -206,7 +221,7 @@ - (void)syncPostsOfType:(NSString *)postType
                    success:^(NSArray *posts) {
                        BOOL hasMore = ([posts count] < PostServiceNumberToFetch) ? NO : YES;
                        [self.managedObjectContext performBlock:^{
-                           [self mergePosts:posts ofType:postType withStatuses:postStatus forBlog:blog purgeExisting:YES completionHandler:^{
+                           [self mergePosts:posts ofType:postType withStatuses:postStatus byAuthor:authorID forBlog:blog purgeExisting:YES completionHandler:^{
```

Comment:
I think `purgeExisting:YES` here might result in wasted bandwidth. To see what I mean test with a very active blog where you posted recently. In my case:

- Open posts list, see a bunch of posts
- Tap on Everyone, post list refresh, purges all posts older than X
- Tap on Only Me, one post is shown, list refreshes and more are loaded

I only have examples on internal sites, I can send you a video if you want.

I think this would work if the filter was author A, author B,... but since we're switching between author:A and author:nil, it's not. When `author:nil`, this will act as before the author filters and clear any post older than the last 40.